client/createResource_test.go

```
@@ -61,6 +62,29 @@
 				},
 			}),
 		},
+		{
+			name: "test link not found callback",
+			args: args{
+				deviceID: deviceID,
+				href:     "/link/not/found",
+				body:     test.MakeSwitchResourceDefaultData(),
+				opts: []client.CreateOption{
+					client.WithDiscoveryConfiguration(core.DefaultDiscoveryConfiguration()),
+					client.WithLinkNotFoundCallback(func(links schema.ResourceLinks, href string) (schema.ResourceLink, error) {
```

golangci-lint

[warning] unused-parameter: parameter 'href' seems to be unused, consider removing or renaming it as _

(revive)